lib/ssz_ex.ex

```diff
@@ -112,35 +112,36 @@ defmodule LambdaEthereumConsensus.SszEx do
 
   @spec hash_tree_root(list(), {:list, any, non_neg_integer}) ::
           {:ok, Types.root()} | {:error, String.t()}
-  def hash_tree_root(list, {:list, type, size}) do
-    if variable_size?(type) do
-      # TODO
-      # hash_tree_root_list_complex_type(list, {:list, type, size}, limit)
-      {:error, "Not implemented"}
+  def hash_tree_root(list, {:list, type, _size} = schema) do
+    limit = chunk_count(schema)
+    len = length(list)
+
+    if basic_type?(type) do
+      pack(list, schema)
     else
-      packed_chunks = pack(list, {:list, type, size})
-      limit = chunk_count({:list, type, size})
-      len = length(list)
-      hash_tree_root_list_basic_type(packed_chunks, limit, len)
+      list_hash_tree_root(list, type)
     end
+    |> hash_tree_root_list(limit, len)
   end
 
   @spec hash_tree_root(list(), {:vector, any, non_neg_integer}) ::
           {:ok, Types.root()} | {:error, String.t()}
-  def hash_tree_root(vector, {:vector, type, size}) do
-    if variable_size?(type) do
-      # TODO
-      # hash_tree_root_vector_complex_type(vector, {:vector, type, size}, limit)
-      {:error, "Not implemented"}
+  def hash_tree_root(vector, {:vector, type, _size} = schema) do
+    if basic_type?(type) do
+      pack(vector, schema)
     else
-      packed_chunks = pack(vector, {:list, type, size})
-      hash_tree_root_vector_basic_type(packed_chunks)
+      list_hash_tree_root(vector, type)
     end
+    |> hash_tree_root_vector()
   end
 
-  @spec hash_tree_root_list_basic_type(binary(), non_neg_integer, non_neg_integer) ::
-          {:ok, Types.root()} | {:error, String.t()}
-  def hash_tree_root_list_basic_type(chunks, limit, len) do
+  def hash_tree_root_vector(chunks) do
```

Comment:
We have to check that the vector is the right size

Reply:
Something like this

```
    if chunks_len > limit do
      {:error, "chunk size exceeds limit"}
```

??

Reply:
More like `chunks_len != limit`, since vectors are of fixed size

Reply:
I don't think that's needed for vector. It doesn't have `limit`.

Reply:
The vector has a `size`. We should check that `length(vector) == size`, just for consistency, since we don't have a way to check that at the type level

Reply:
Gotcha